pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -202,6 +204,34 @@ func getVMIMachine(vmi *k6tv1.VirtualMachineInstance) (guestOSMachineType string
 	return
 }
 
+func getVMIPod(vmi *k6tv1.VirtualMachineInstance) string {
+	if vmi == nil {
+		return ""
+	}
+
+	var vmiPodName string
+	for _, obj := range kvPodInformer.GetStore().List() {
+		pod, ok := obj.(*k8sv1.Pod)
+		if !ok {
+			continue
+		}
+
+		if pod.Namespace == vmi.Namespace && pod.Labels["kubevirt.io/created-by"] == string(vmi.UID) {
+			if vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed {
+				if pod.Name == vmi.Status.MigrationState.TargetPod {
+					vmiPodName = pod.Name
+					break
+				}
+			} else {
+				vmiPodName = pod.Name
+				break
+			}
+		}
+	}
```

Comment:
Using the namespace index you will reduce the pod count and save the namespace check.

Also, I don't think the `migrationState.Completed` is enough. The migration state is also completed for failed migrations, and in that case, the correct pod is still the source pod.

```go
	objs, err := kvPodInformer.GetIndexer().ByIndex(cache.NamespaceIndex, vmi.Namespace)
	if err != nil {
		return ""
	}
	for _, obj := range objs {
		pod, ok := obj.(*k8sv1.Pod)
		if !ok {
			continue
		}

		if pod.Labels["kubevirt.io/created-by"] == string(vmi.UID) && pod.Status.Phase == k8sv1.PodRunning {
			if vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed && !vmi.Status.MigrationState.Failed {
				if pod.Name == vmi.Status.MigrationState.TargetPod {
					vmiPodName = pod.Name
					break
				}
			} else {
				vmiPodName = pod.Name
				break
			}
		}
	}
```	
Generally I think we can be more conservative and rely on the nodename of the running pod and the vmi nodename:
```go
	objs, err := kvPodInformer.GetIndexer().ByIndex(cache.NamespaceIndex, vmi.Namespace)
	if err != nil {
		return ""
	}
	for _, obj := range objs {
		pod, ok := obj.(*k8sv1.Pod)
		if !ok {
			continue
		}

		if pod.Labels["kubevirt.io/created-by"] == string(vmi.UID) && pod.Status.Phase == k8sv1.PodRunning {
			if vmi.Status.NodeName == pod.Spec.NodeName {
				vmiPodName = pod.Name
				break
			}
		}
	}
```

Reply:
good idea! I tried to rely on namespace but it didn't work, trying with the node

Reply:
the namespace is the same in both virt-launcher, but different node (until and if we allow migrating to the same node). In the latter case probably we should rely on the creationtimestamp

Reply:
@fossedihelm 2 questions, if using the node, no need to the MigrationState.TargetPod?
```
 if vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed && !vmi.Status.MigrationState.Failed {
				if pod.Name == vmi.Status.MigrationState.TargetPod {
					vmiPodName = pod.Name
```

and what do you mean rely on the creationtimestamp? you mean the pod creationtimestamp?

Reply:
fixed the code, not sure it's what you meant

Reply:
fixed the code and verified it's working, not sure it's what you meant.
now adding unit tests

Reply:
if migration can't be on the same node I agree. fixed and also added unit tests.
thanks a lot (: